src/main/java/com/metamx/http/client/response/InputStreamResponseHandler.java

```diff
@@ -31,17 +30,18 @@
   @Override
   public ClientResponse<AppendableByteArrayInputStream> handleResponse(HttpResponse response)
   {
+    ByteBuf content = response instanceof HttpContent ? ((HttpContent) response).content() : Unpooled.EMPTY_BUFFER;
     AppendableByteArrayInputStream in = new AppendableByteArrayInputStream();
-    in.add(getContentBytes(response.getContent()));
+    in.add(content.array());
```

Comment:
Is there a guarantee that the array exists? ByteBuf has `hasArray()` method. Also `arrayOffset()` should be considered.